File: TeamCubing.API/ClientApp/src/app/models/roomSolve.ts
export interface IRoomSolve {
  id: number;
  solveNumber: number;
  results: ISolveResult[];
  roomId: number;
  scramble: string;
}

export interface ISolveResult {
  id: number;
  roomSolveId: number;
  userName: string;
  time: number;
}

File: TeamCubing.DAL/Models/RoomSolve.cs
namespace TeamCubing.DAL.Models;

public class RoomSolve
{
    public int Id { get; set; }

    public int SolveNumber { get; set; }

    public string Scramble { get; set; }

    public DateTime StartTime { get; set; }

    public int RoomId { get; set; }

    public Room Room { get; set; }

    public List<RoomSolveResult> Results { get; set; }
}
